### PR TITLE
coremidi input port and autoconnect

### PR DESCRIPTION
**Pull request by ColinKinloch**
Adds a port to the coremidi driver by default and optionally automatically connects this to all midi sources (related to #414).

### CODE REVIEW

**Comment by derselbst** [comment]
Oh, very nice. Thank you.

Could you please explain why you are creating this midi input port? Or what's the difference to `MIDIDestinationCreate()`? Sry, I don't have any experience with CoreMidi and a quick reading didn't cleared it up, so just asking.

**Comment by ColinKinloch** [comment]
To be honest, I don't know. 

I've never been able to get fluidsynth to interact with a MIDI keyboard on macOS before now.

I spent a while searching the web and looking through the CoreMIDI docs without much luck.

CoreMIDI via `MIDIGetNumberOfSources` and `MIDIGetSource` allows you to enumerate input devices (virtual / hardware) which, like the output of `MIDIDestinationCreate`, are "endpoints".

`MIDIPortConnectSource` allows a client to connect a source endpoint to a "port". This doesn't require a destination for the client to exist and there doesn't seem to be an analogue in CoreMIDI for connecting destinations and sources.

**Comment by mawe42** [comment]
I dont' know CoreMidi either, but from reading some explanations I think it works like this: 

MIDIDestinationCreate` creates a virtual MIDI device that other applications can connect to. So that gets used if you want to use fluidsynth as a virtual instrument in Garageband etc. Garageband will enumerate all destinations (virtual or hardware) known to the system. As soon as you choose one, it creates an output port to send MIDI messages to that destination. So in essence MIDIDestinationCreate is used to open up to connections initiated by other apps.

An input port created with `MIDIInputPortCreate` is simply the receiving end of a connection to a source, be it virtual or hardware. So that is used when we ourselves initiate the connection to a MIDI source and want to receive events from it.

Anyway, as I said... just a wild guess. CoreMIDI has some strange concepts when compared to the Linux world of things.

File: src/drivers/fluid_coremidi.c [hunk 1]
```diff
@@ -67,6 +69,31 @@ void fluid_coremidi_driver_settings(fluid_settings_t *settings)
     fluid_settings_register_str(settings, "midi.coremidi.id", "pid", 0);
 }
 
+static void fluid_coremidi_autoconnect(fluid_coremidi_driver_t *dev, MIDIPortRef input_port)
+{
+    unsigned int i;
+    int source_count = MIDIGetNumberOfSources();
+    for(i = 0; i < source_count; ++i)
```
Comment (derselbst) [inline src/drivers/fluid_coremidi.c]: Now, the  comparision `i < source_count` differs in signedness. Prefer using int.

**Pull request merged**